packages/web3-eth-accounts/src/scrypt.js

```diff
@@ -0,0 +1,71 @@
+var scryptsy = require('scryptsy');
+
+var scrypt;
+
+var isNode = Object.prototype.toString.call(typeof process !== 'undefined' ? process : 0) === '[object process]';
+if (isNode) {
+    var NODE_MIN_VER_WITH_BUILTIN_SCRYPT = '10.5.0';
+    var NODE_MIN_VER_INCOMPAT_SCRYPT_PKG = '12.0.0';
+    var semver = require('semver');
```

Comment:
That is quite painful in webpack I fear... One can have webpack and node (e.g. lambda functions).

Reply:
Can you explain a bit more?  If this broke something in a node FaaS context, I will be glad to work on a fix.

Reply:
I am just not sure whether or not webpack correctly supports this dynamic require. Needs to be tested.

Reply:
It doesn't. Webpack fails at dynamic require at web3@1.2.1. Any idea how to workaround it?
![image](https://user-images.githubusercontent.com/3285000/62639329-fa9e3680-b947-11e9-8579-348e72442b0f.png)

Reply:
I've seen the warning too, but it didn't cause my build to fail, and my build worked as expected. Are you sure you actually got a failed build?

In any case, the `web3-eth-accounts` module should be refactored to make use of `"browser"` in `package.json` instead of using runtime `isNode` detection. I'll work on that very soon.

Reply:
As I said in my previous comment, this should be refactored. But in the meantime I don't think the warning will result in broken builds. Here's what I tried:

```
$ cd ~/temp
$ npx create-react-app foo
$ cd foo && yarn add web3
```
After that I edited `src/App.js` to `import Web3 from 'web3'` and I set `window.Web3 = Web3`. Then:
```
$ yarn start
```
Once the dev server was available at localhost:3000 I was able to use `window.Web3` without any problems, though I did see the webpack warning in the console output of `yarn start`.

I checked that I could use `web3.eth.accounts` to create/encrypt/decrypt — worked as expected, so while the webpack warning isn't nice, it doesn't seem to prevent browser builds (w/ webpack) from working.

I'm not 100% sure about webpack builds targeting Node.js; but shouldn't be a problem because such builds couldn't pack up the `scrypt` package in any case (it's a compiled thing).

Reply:
Hey @michaelsbradleyjr you are absolutely right. I had other issue and `critical dependency` got me triggered. Thanks for detailed answer and check. Really appreciated.